pkg/metricshandler/metrics_handler.go

```diff
@@ -208,7 +209,7 @@ func (m *MetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	m.metricsWriters = metricsstore.SanitizeHeaders(m.metricsWriters)
+	m.metricsWriters = metricsstore.SanitizeHeaders(string(gotContentType), m.metricsWriters)
```

Comment:
Can we have a boolean here "isOpenMetricsFmt" instead? I think on txt format we would want to fall back as well.

Reply:
Enabling native histograms sets the `scrape_protocols` to `[ PrometheusProto, OpenMetricsText1.0.0, OpenMetricsText0.0.1, PrometheusText0.0.4 ]`. [`ProtoFmt`](https://github.com/prometheus/common/blob/658f673b7580e72a7d0c0fac26a67858b782fbd5/expfmt/expfmt.go#L25) should catch all [`Proto` formats](https://github.com/prometheus/prometheus/blob/69abd6d9f65a6358025461510237366f1d939c70/config/config.go#L451), including any [`Proto`-text ones](https://github.com/prometheus/common/blob/658f673b7580e72a7d0c0fac26a67858b782fbd5/expfmt/expfmt.go#L33-L35) as well.

I have restricted this behavior to "only Prometheus-proto" rather than "anything other than OpenMetricsText" so we can be sure if anything other than these two starts causing problems for us in the future, and we don't end up `s/info|statesets/gauge` for it since that may be due to an entirely different bug.

Reply:
Thanks for the clarification! I misunderstood, that makes sense to me. :)